src/controllers/auth.controller.js

```diff
@@ -0,0 +1,185 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { User } from '../modules/User.js';
+import { jwtService } from '../service/jwt.service.js';
+import { tokenService } from '../service/token.service.js';
+import { userService } from '../service/user.service.js';
+import bcrypt from 'bcrypt';
+import { ConsoleLogger } from '../untils/consoleLogger.js';
+
+// function validateEmail(value) {
+//   const EMAIL_PATTERN = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+//   if (!value) {
+//     return 'Email is required';
+//   }
+
+//   if (!EMAIL_PATTERN.test(value)) {
+//     return 'Email is not valid';
+//   }
+// }
+
+// function validatePassword(value) {
+//   if (!value) {
+//     return 'Password is required';
+//   }
+
+//   if (value.length < 6) {
+//     return 'At least 6 characters';
+//   }
+// }
+
+const register = async (req, res) => {
+  const { email, password, userName } = req.body;
+
+  const errors = {
+    email: userService.validateEmail(email),
+    password: userService.validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Email, password, are not validate', errors);
+  }
+
+  const hashPass = await bcrypt.hash(password, 10);
+
+  await userService.register(email, hashPass, userName);
+
+  res.status(201).send({
+    message:
+      'User registered successfully, check your email for activate account',
+  });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+  await user.save();
+
+  res.send(user);
+};
+// як перенаправити на профіль????
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await User.findOne({ where: { email } });
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  generateToken(res, user);
```

Comment:
The `generateToken` function is used here but is not defined within the same scope. Ensure that `generateToken` is defined and accessible in this context, or import it if it's defined elsewhere.